engine/server/engine/enclave_manager/enclave_pool.go

```diff
@@ -203,34 +242,58 @@ func (pool *EnclavePool) destroyPreviousIdleEnclaves() error {
 }
 
 // run method is in charge of filling the enclave pool channel with new idle enclaves
-func (pool *EnclavePool) run() {
+func (pool *EnclavePool) run(ctx context.Context) {
 
-	defer close(pool.idleEnclavesChan)
+	defer func() {
+		close(pool.idleEnclavesChan)
+	}()
 
-	fails := 0
+	// init the pool
+	if err := pool.fill(ctx); err != nil {
+		logrus.Errorf("An error occurred filling the enclave pool. Error\n%v", err)
+		return
+	}
 
 	for {
-		newEnclaveInfo, err := pool.createNewEnclave()
-		if err != nil {
-			// Retry strategy if something fails
-			logrus.Errorf("An error occurred creating a new idle enclave. Error\n%v", err)
-			fails++
-			if fails >= maxRetryAfterFails {
-				logrus.Debugf("The enclave pool sub-routine has been canceled after reaching the max retries value '%v'", maxRetryAfterFails)
+		// wait until receive the re-fill signal or the close signal
+		select {
```

Comment:
YES, now your `for ... select` loop makes a lot more sense, it waits on both channels and the first one to receive a value enters the clause. That's much better